community/kernel/src/main/java/org/neo4j/kernel/impl/proc/ReflectiveProcedureCompiler.java

```diff
@@ -342,12 +359,7 @@ private CallableUserFunction compileFunction( Class<?> procDefinition, MethodHan
     private CallableUserAggregationFunction compileAggregationFunction( Class<?> definition, MethodHandle constructor,
             Method method, QualifiedName funcName ) throws ProcedureException, IllegalAccessException
     {
-        if ( funcName.namespace() == null || funcName.namespace().length == 0 )
-        {
-            throw new ProcedureException( Status.Procedure.ProcedureRegistrationFailed,
-                    "It is not allowed to define functions in the root namespace please use a namespace, " +
-                            "e.g. `@UserFunction(\"org.example.com.%s\")", funcName.name() );
-        }
+        restrictions.verify( funcName );
```

Comment:
Nice cleanup.

Reply:
I didn't do it for cleanup purposes - I did it so that I could turn the check off for built-in functions...

Reply:
yes I know, the cleanup was collateral.